returnn/datasets/util/vocabulary.py

```diff
@@ -123,6 +130,15 @@ def create_vocab_dict_from_labels(cls, labels):
     assert len(d) == len(labels), "some labels are provided multiple times"
     return d
 
+  @classmethod
+  def create_vocab_from_labels(cls, labels):
+    if len(labels) < 1000 and all([len(label) == 1 for label in labels]):
+      # are these actually ordered raw bytes? -> assume utf8
+      if all([ord(label) <= 255 and ord(label) == idx for idx, label in enumerate(labels)]):
+        return Utf8ByteTargets()
```

Comment:
One potential issue is that this does not support datasets anymore that use utf8 bytes that are not ordered. We could fix that by changing Utf8ByteTargets to use explicit labels/vocab.

But I don't think that anybody is using something like that, so it may be fine as it currently is.

Reply:
But why would it not be ordered, or why would you use `Utf8ByteTargets` then? `Utf8ByteTargets` is intended to represents the bytes, so it is by definition exactly the byte.

If we ever would have or need such a thing, I think we should just create a separate vocab class.

And the user would just not use this fallback code for that but instead properly define the vocab.

So I don't think this is a problem.